ckanext/satreasury/plugin.py

```diff
@@ -251,69 +252,36 @@ def get_helpers(self):
         }
 
     # IDomainObjectModification
-
     def notify(self, entity, operation):
-        if build_trigger_enabled():
-            if build_already_queued():
+        if travis.build_trigger_enabled():
+            pending_builds = travis.get_queued_builds()
+            if pending_builds:
                 log.info("Not triggering build because already queued")
+                show_success_message_for_build(pending_builds[0])
             else:
                 if isinstance(entity, model.Package) and entity.owner_org:
-                    trigger_build()
+                    try:
+                        created_request = travis.trigger_build()
+                    except requests.exceptions.HTTPError as e:
+                        ckan_helpers.flash_error("An error occurred when updating the static site data. Technical details: %s" % e.message)
+                        return
+
+                    # Sleep a while to wait for the build to be created
+                    time.sleep(2)
```

Comment:
oh no - we can't rely on a sleep like this. Is this needed because the trigger request just creates a "request" and doesn't mean the build is necessarily there?

In that case we should just link to the builds list, because there's no way of knowing that the build will really be created in 2 seconds - so when Travis is very busy we might show an error when it's just taking a little while

Reply:
Yeah, I just put it there in case the build isn't created immediately. Link to list of builds could be better. Otherwise we need to poll the api until a the build is created.